FirebaseAI/Sources/GenerativeModel.swift

```diff
@@ -86,13 +88,15 @@ public final class GenerativeModel: Sendable {
        toolConfig: ToolConfig? = nil,
        systemInstruction: ModelContent? = nil,
        requestOptions: RequestOptions,
-       urlSession: URLSession = GenAIURLSession.default) {
+       urlSession: URLSession = GenAIURLSession.default,
+       useLimitedUseAppCheckTokens: Bool = false) {
```

Comment:
This could be done in a follow-up since it's an internal implementation detail but could we consider putting this value in [`FirebaseInfo`](https://github.com/firebase/firebase-ios-sdk/blob/e5962228d77cf20c2795a65d9ad85f91aa3e1082/FirebaseAI/Sources/FirebaseInfo.swift) instead? That way we won't need to plumb it through to `GenerativeModel`, `ImagenModel` and `GenerativeAIService` separately.

Just double checking, @paulb777 does this align with the types of values you consider reasonable for `FirebaseInfo`?

Reply:
I thought the same thing (see [this thread](https://github.com/firebase/firebase-ios-sdk/pull/15099#discussion_r2195574606)), but since we don't do the same with `APIConfig`, I assumed it wasn't preferable. But I'm fine either way